Refreshed recipe and updated to v2.1.1

pelson opened the pull request:
Checklist:

* [X] Used a fork of the feedstock to propose changes
* [X] Ensured the license file is being packaged.

pelson committed:
Refreshed recipe and updated to v2.1.1

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found some lint.

Here's what I've got...


For **recipe**:

 * Recipe maintainer "dkillick" does not exist

pelson commented:
@conda-forge-admin, please rerender

pelson committed:
Remove maintainer.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

pelson commented:
@conda-forge-admin, please rerender

conda-forge-admin committed:
MNT: Re-rendered with conda-build 3.17.8, conda-smithy 3.2.12, and conda-forge-pinning 2019.01.29

ocefpaf commented on recipe/meta.yaml:
```diff
@@ -37,20 +37,17 @@ test:
-  home: http://www.scitools.org.uk/iris
-  license: LGPL-3.0
+  home: https://scitools.org.uk/cf-units/docs/latest/
+  license: LGPLv3
```
@pelson this is not a big deal and you don't have to revert it but we are trying to consolidate the licenses strings around the [SPDX](https://spdx.org/licenses/LGPL-3.0.html) standard.

ocefpaf commented:
@pelson do you know if the Python 2.7 failures here are real or if we need to adjust the tests? (I'm not really following the changes upstream lately.)

pelson commented:
They are kind of genuine. There are files in cf-units that are not python 2 compatible (the new TeX feature) - but I made sure I followed a well trodden path to ensure that those files never get imported in python 2 (else you get the SyntaxError we are seeing).

I'm actually going to improve the way the tests are done in this feedstock, as currently they are using the tests defined in the source repository on the package installed in site-packages, which is a bit weird IMO.

pelson committed:
Improve the testing by using the installed test suite, not the source repository test suite.

pelson commented:
Looks like I've fixed what we were seeing, but now another (genuine?) test failure:

```
    def test_read_xml_invalid_path(self):
        with self.assertRaises(_ud.UdunitsError) as cm:
            _ud.read_xml(b'/not/a/path.xml')
        ex = cm.exception
    
        self.assertEqual(ex.status_msg(), 'UT_OPEN_ARG')
>       self.assertEqual(ex.errnum, errno.ENOENT)
E       AssertionError: 0 != 2
```

From what I can tell, this test was updated a few months ago in https://github.com/SciTools/cf-units/pull/116/files, and added a few years ago. For now, I suggest we skip this test on windows, and raise the issue on the cf-units tracker. Interestingly, this test is also *only* failing on Win and py27.

pelson committed:
Added a skipper for the failing windows py2 test.

ocefpaf commented:
Thanks Phil! I'm looking forward to play with the new latex functionality.

pelson commented:
🎉 

> Thanks Phil! I'm looking forward to play with the new latex functionality.

Yay! I really like the idea of adding awesome functionality in python 3 only. 👍
Let me know if you come up with any issues with it (beyond the fact that we didn't implement Log units yet). Also, FWIW I've essentially implemented a UDUNITS2 parser - in theory we could consider submitting a change to the CF spec to talk about a units grammar, rather than mentioning a particular implementation (UDUNITS2) of said grammar. One for the future...
